### ISSUE TITLE
Fix margin/padding for messages on homepage (Basis)

### ISSUE THREAD

**Issue by laryn** [backdrop/backdrop-issues#5769]
## Description of the bug

I was testing out the preview release and I think we can do better on the placement of the messages on the homepage:

![CleanShot 2025-01-06 at 08 00 10@2x](https://github.com/user-attachments/assets/3ac93e1c-ff15-4c2c-94f9-cad11c0ab461)

We now have a way forward to make backwards-compatible CSS fixes using the [supplemental CSS](https://github.com/backdrop/backdrop-issues/issues/4782) selectors. Consensus seems to be around fixing the strange gaps that are so obvious upon first installation and subsequent messages in the default placements with a minimal fix. If issues that may show up when messages are moved to other regions are also resolved it's obviously a bonus -- but not the primary goal of this issue, and should not delay things. If at all possible we want this bug fix to get into 1.30.0.

**Comment by stpaultim** [comment]
Yeah, I've been thinking about this too. I'm just not sure about what the best alternative is yet. However, I agree that the current situation is kinda ugly.

**Comment by olafgrabienski** [comment]
> I think it would be better below the cards ...

On smartphones and on many tablets messages wouldn't be visible at first sight, if they were below the cards.

Let's try to keep the messages (more or less) at their place and to remove the white gap. Maybe style the messages block with `position: absolute`? Something like this:

```
.front .block-system-messages {
  position: absolute;
  top: 22rem;
  right: 0;
  left: 0;
  z-index: 2;
}
```

The example could certainly be improved, e.g. the `top` value should be adapted for phones. Here's however a screenshot to illustrate the idea:

![backdrop-home-message-suggestion](https://user-images.githubusercontent.com/14188307/189883420-0d443be5-6cab-43e3-8604-7735fc8bbe03.png)

**Comment by stpaultim** [comment]
Oh, I think I found an issue I could work on and would love to see get fixed. This one has been bothering me for a long time (since we added the cards). Now that it has been brought up again in https://github.com/backdrop/backdrop-issues/issues/6074, I think it's time to fix it. 

It's not a good first look for Backdrop. I'll try to submit a PR for this soon, but please - someone go ahead and beat me to it. I'd volunteer to advocate for this, but it it's not a new feature, so it really doesn't need an advocate.

**Comment by olafgrabienski** [comment]
This issue came up again in #6220, reported by @alanmels. Now it's really time to fix it, I guess! @stpaultim Are you still up to work on it?

**Comment by wesruv** [comment]
Messing around a bit I considered a few approaches:
* Rearranging so the status messages are above the hero, but I don't think that helped anything
* Position absolute (like @olafgrabienski suggested), but in practice that can open up some other issues

I think the most straightforward way to address this would be a small design adjustment to remove the gray bg behind the cards, and using a drop shadow to distinguish the cards from the white bg. I think it's the least disruptive and straightforward solution.
![image](https://github.com/backdrop/backdrop-issues/assets/5607236/a87d8b2e-361f-4b5b-b555-9648ca535850)

**Comment by jenlampton** [comment]
I'm so happy we're working on this issue - it bothers me so much with every install!

Unfortunately, we can't change any front-end CSS in any way that would affect existing Backdrop sites -- understanding that most backdrop sites that are running today have already fixed this problem, and if we fix it differently than they dif, we're going to break those live sites 😱 

We've been struggling with this problem since the very beginning of Backdrop, and have a few ideas about how we can fix CSS for new sites without breaking existing sites. @wesruv  I'd love to have your opinion on the current "leading" idea in https://github.com/backdrop/backdrop-issues/issues/4782

**Comment by bugfolder** [comment]
Just throwing out ideas here. Add a checkbox (or set of checkboxes?) to the Basis theme:

[ ] Use Basis2024 settings

When updating from old sites, default is unchecked. When installing new sites, default is checked. People with existing sites will have nothing break, but they can try out the new settings if they're curious.

In a couple of years, we can add another checkbox for "Use Basis2027 settings".

**Comment by stpaultim** [comment]
Here is the "Meta" issue for deciding upon the best strategy for css improvements. 
https://github.com/backdrop/backdrop-issues/issues/4167

**Comment by jenlampton** [comment]
@bugfolder I believe that was discussed already in the other issues, but I'd recommend giving them a read-through and adding similar ideas there. Let's not distract too much from the solution for messages, being determined here :)

**Comment by bugfolder** [comment]
Yes, i see the other issue from @stpaultim now.

**Comment by olafgrabienski** [comment]
Back to @wesruv's suggestion: Looking at the screenshot, I like the design adjustment to remove the gray background and using a drop shadow instead. What do others think?

PS: I wanted to test the PR to see how the page looks without messages, but there isn't a sandbox site.

**Comment by stpaultim** [comment]
@olafgrabienski The sandbox is working for me now. 

In addition to the issue about making changes to CSS raised by @jenlampton, this would only solve the problem with Basis, right out of the box. It doesn't solve the problem for other themes or other configurations. 

Having said that, I think it would be OK to just solve this for Basis out of the box for now. The most important problem for me, is fixing the appearance of the first thing anyone sees when they load Backdrop. If this does not work in other themes or after config changes, the "Page Messages" block can be moved or the Nice Messages (https://backdropcms.org/project/nicemessages) contrib module can be installed. 

So the solution by @olafgrabienski would be more flexible, but I understand that Wes anticipates other issues. Wes's solution would be fine by me, for now, if we can do it in a backward compatible way.

**Comment by olafgrabienski** [comment]
Thanks @stpaultim! I was able to have a look at the page without messages now. Here's a screenshot:

![image](https://github.com/backdrop/backdrop-issues/assets/14188307/1d4d540e-c164-4dcb-ae0a-a547dbe93324)

Looks good, in my opinion. Maybe the cards block could need some more top padding, though, e.g. `3rem` instead of `1rem`.

**Screenshot 3rem padding-top:**
![image](https://github.com/backdrop/backdrop-issues/assets/14188307/591b0d93-a7bc-4434-8a70-43f761e0d680)

**Comment by stpaultim** [comment]
> Ok, I see the PR is adding shadowing to the cards and has nothing to do with the messages!

@izmeez The PR removed the background image from the "cards" region and adds a shadow behind the cards, to fix a display problem with messages. 

See this before and after. You can recreate this problem by clearing the caches on a demo site and the PR sandbox.

The white space that appears with messages behind the hero block and the cards block looks broken. This PR makes it look better.

![image](https://github.com/user-attachments/assets/5eaac70d-6c3c-44ab-853f-41353795573f)

![image](https://github.com/user-attachments/assets/e54aa754-5c26-4e78-bfaf-d68a444c16f4)

**Comment by stpaultim** [comment]
I've added a second PR that pretty much follows the example by @wesruv, except limited the changes to sites with the .update-1-30 class.

There is one additional change to address a very minor issue, which may or may not be worth addressing. 

I tried moving the messages block around a bit within the existing home page layout and found that in this position, the spacing could be better. 

PR by @wesruv  - 
![image](https://github.com/user-attachments/assets/888259f8-0095-4d4c-9520-312ba66d828a)

Adding this `.update-1-30 .l-top .block-system-messages:first-child` makes that a bit nicer. I recognize that this is a very minor issue in a fringe situation, so this extra line in the css may not be necessary. 

![image](https://github.com/user-attachments/assets/5bc619a1-388e-4691-89b1-70c19d233f88)

To test this PR, I think it's good to move around the default blocks on the home page layout and flush the cache and see what happens.

**Comment by avpaderno** [comment]
I tested the PR on the preview site, by changing the order of the blocks used by the *Home page* layout, even putting the cards view as first block in the *Top* region. I did not find any issue; the blocks were rendered as expected.

**Comment by olafgrabienski** [comment]
@stpaultim Thanks a lot for the PR! It works for me in the sandbox.

I _think_ by the way, your screenshots in https://github.com/backdrop/backdrop-issues/issues/5769#issuecomment-2564558784 are misleading. Out of the box, the _Page messages_ block isn't above the _Welcome_ hero block, but below (as in the following screenshot):

![image](https://github.com/user-attachments/assets/81a4f756-b4fd-488d-9e1d-a572003ad5a0)

**Comment by stpaultim** [comment]
> I think by the way, your screenshots in https://github.com/backdrop/backdrop-issues/issues/5769#issuecomment-2564558784 are misleading. Out of the box, the Page messages block isn't above the Welcome hero block, but below (as in the following screenshot):

@olafgrabienski Yes, I will edit the text there, but those screenshots were put there to help explain a css change that I made. It is only what happens if the messages block is moved. I am not sure if this is a problem we needed to fix, but it is a problem that I chose to fix in the PR.

**Comment by olafgrabienski** [comment]
@stpaultim – thanks for the clarification!

> How much should we worry about the appearance of this block in positions OTHER than the default position?

Not very much, in my opinion. The issue is primarily the out of the box position, and it's the first thing you see after the installation of Backdrop. However, if it's an easy fix, why not? (I'm going to left a comment in the PR.)

**Comment by quicksketch** [comment]
@stpaultim Yay, the messages look so much better now!! I hope this isn't too tricky to figure out, but the outline around the messages block when using contextual links is missing the bottom outline:

![image](https://github.com/user-attachments/assets/d00f9cd1-1f5d-43b0-860f-a76c4bf4415b)

**Comment by stpaultim** [comment]
@quicksketch The problem seems to be that the "cards" view has a white background that is covering the bottom border of the contextual message block. I believe that this is the expected behavior, when the background is anything other than white. But, because of the white background it looks a bit strange. 

I fixed it with a top-margin of 1px on that specific view. I believe that this is an acceptable fix, because it ONLY effects this view and because this view requires that the background always be white. With a white background, adding 1px to the top seems pretty safe. 

PR updated with this fix.

**Comment by quicksketch** [comment]
As discussed during today's dev meeting:

- Revert the 1px margin change.
- Handle in a separate issue using `outline-offset`. See https://github.com/backdrop/backdrop-issues/issues/6802

Because this is a global issue with contextual links and not specifically related to the messages or cards blocks.

**Comment by jenlampton** [comment]
Removing RTBC label then?

**Comment by alanmels** [comment]
Hard to believe this has been open since September 7, 2022, and it’s finally getting resolved. Kudos to everyone for sticking with it!

**Comment by jenlampton** [comment]
I love it :) 
<img width="1631" alt="Screenshot 2025-01-02 at 7 27 48 PM" src="https://github.com/user-attachments/assets/16bb3277-94c8-44ba-8821-59983bfdff59" />

I have filed an alternate PR at https://github.com/backdrop/backdrop/pull/4973 that removes the 1px contextual link fix.

**Comment by jenlampton** [comment]
uhoh, I think I found a problem. If you delete the hero block, the menu gets covered up :( 
<img width="1630" alt="Screenshot 2025-01-02 at 7 43 23 PM" src="https://github.com/user-attachments/assets/8a32fe5e-8306-4907-9744-9d092128281e" />

**Comment by stpaultim** [comment]
@jenlampton Seems like you are working on this. Go ahead.

This -2rem issue is related to this. https://github.com/backdrop/backdrop-issues/issues/4095

I am very eager to get issue 4095 solved. There is a 2 rem bottom margin in Basis that is causing lots of trouble. My PR in 4095 is designed to solve that. I have a hunch that these two PR's together might work. 

But, you will probably find another solution.

**Comment by jenlampton** [comment]
I posted an updated PR at https://github.com/backdrop/backdrop/pull/4974, which is actually less CSS change than the previous one: 
<img width="1092" alt="Screenshot 2025-01-02 at 8 10 33 PM" src="https://github.com/user-attachments/assets/f565f3f3-8dd2-44fb-82fa-3df2c873da24" />

This solution solves the issue I just encountered, and also I think addresses @olafgrabienski's concern from an earlier PR, without needing any new css for individual blocks.

**Comment by stpaultim** [comment]
I see you just posted PR. I'm going to look and see how it effects that other issue.

**Comment by jenlampton** [comment]
Okay, I've updated my latest PR to ALSO include a change for the messages block on the home page.  With messages the home page now looks like this: 
<img width="1635" alt="Screenshot 2025-01-02 at 9 32 23 PM" src="https://github.com/user-attachments/assets/e538d978-a3a4-4807-b468-78071f994c18" />

Without, the page looks like this: 
<img width="1637" alt="Screenshot 2025-01-02 at 9 32 01 PM" src="https://github.com/user-attachments/assets/1684818e-e117-47e3-ac35-55dc050b1d65" />

**Comment by stpaultim** [comment]
@jenlampton 

You seem to have removed some padding below the message. Isn't this a little too tight? There was more space here before.
![image](https://github.com/user-attachments/assets/8e973011-95e1-4725-91ae-0d9c02081e03)

**Comment by stpaultim** [comment]
Thanks @jenlampton for coming up with a better solution to this problem. 

I'm ok with PR as it currently is, but I think you might have taken a bit too much padding away from the bottom of the Messages block (see my previous comment). 

Unlike the previous solution, this solution adds a constant light background color to the entire `top` region. I'm fine with this solution. This is closer to our original design idea with the cards layout AND sets the top region off slightly from the "content" region.

**Comment by olafgrabienski** [comment]
> taken a bit too much padding away from the bottom of the Messages block

I agree, but only for the home page. There, a space of `1em` below the Messages block would be nice – especially when the message are moved above the Hero block, but it looks also better when the messages are in their default position between Hero and Cards.

**Comment by indigoxela** [comment]
I have a heretic suggestion...

Initially the whole .l-top got the grey background, but that caused a regression, which we fixed back in 2022 - introducing another regression (the messages gap problem), which gets reported regularly.

Some time has passed, some silly browser software passed away, and we have a new approach available: `has()`.

So, adding the following code makes the .l-top background grey, if it contains the cards block:

```
.front.update-1-30 .l-top:has(.block-views-promoted-cards-block) {
  background-color: #f2f2f2;
}
```
A 3-liner. :grin: 

I think, we can use "has" now. https://caniuse.com/css-has

**Comment by indigoxela** [comment]
Here's an alternative PR demonstrating my idea: https://github.com/backdrop/backdrop/pull/4978

How to test:

- Go to home and flush caches to trigger a message -> messages have the same grey bg as cards
- Go to home and then to "This page / Layout" via admin menu, and move the cards to some other region
- Save (you get redirected back to home) -> top has white bg

**Comment by indigoxela** [comment]
> However, there are problems when messages are moved to the top.

I don't get it... By default, after install, message **are** in the top region. @izmeez where are you testing? Is that an older install, but updated?

**Comment by indigoxela** [comment]
> Messages moved above the hero block.

Is moving messages around something this issue tries to address? Maybe I missed that. I thought, it's about the default placement and what home looks like after install.

This looks weird even without my suggested change, BTW, because regardless of the background color the messages seem to "tear the page".

**Comment by avpaderno** [comment]
The issue summary should be updated, since it suggests moving messages below cards, while the consensus seems to be that the cards background should be changed, which is what the PRs do now.

We are trying to make the messages appear good in the default position the message block has.

**Comment by avpaderno** [comment]
I guess that the only way to avoid what shown in https://github.com/backdrop/backdrop-issues/issues/5769#issuecomment-2571272257 would be to find out which background color is used from the top "block," but I am not sure that is possible without JavaScript nor that we want to implement such code.

**Comment by indigoxela** [comment]
> I guess that the only way to avoid what shown in ... would be to find out which background color is used from the top "block,"

I agree. Catching every possible reordering seems impossible without JavaScript to monitor block (type) order on home. There's not only the background color, but all sorts of margin special-cases to consider. Maybe technically feasible, but AFAIK not topic of this issue. For sure not solvable with a CSS 3-liner. :speak_no_evil:

**Comment by avpaderno** [comment]
Thinking more, it would be possible to avoid that issue, by changing the background colors for the block containing the cards. (I guess there could be issues with other colors, though. 😉)

**Comment by indigoxela** [comment]
> it would be possible to avoid that issue, by changing the background colors for the block containing the cards

I belief, this has already been attempted. With mixed success, as this immediately leads to other visual problems - either cards or promoted content looking lost and/or unstyled...

**Comment by avpaderno** [comment]
Yes, that is definitely the case. 😉

![screenshot](https://github.com/user-attachments/assets/d71199c5-0c3b-4d40-9521-714d2430ef46)

https://github.com/backdrop/backdrop-issues/issues/5769#issuecomment-2012845535 gave a different suggestion, which does not seem it was considered. (If it was, I apologize for not understanding what the decision was.)

**Comment by indigoxela** [comment]
> https://github.com/backdrop/backdrop-issues/issues/5769#issuecomment-2012845535 gave a different suggestion, which does not seem it was considered.

I belief, it has been considered - look at yesterday's screenshots above.

**Comment by avpaderno** [comment]
https://github.com/backdrop/backdrop-issues/issues/5769#issuecomment-2568657332 contains the last screenshot showing shadows for cards, but it does not explicitly say they should be removed, nor why they should be removed.

On https://github.com/backdrop/backdrop-issues/issues/5769#issuecomment-2568713468, the shadows are no longer there.

**Comment by olafgrabienski** [comment]
Yep, until recently, everyone was happy with the first approach: remove the gray background from cards and add box shadows. It seems the PR changed direction when @jenlampton discovered an issue with @stpaultim's PR in https://github.com/backdrop/backdrop-issues/issues/5769#issuecomment-2568657332. With @indigoxela's PR we now have a third approach. Each approach has pros and cons, but given the former consense about the first approach, I'd like to know why we shouldn't continue to improve that one.

**Comment by alanmels** [comment]
Hey everyone, sorry to go a bit off-topic here, but reading through this discussion reminded me of something @al-rozhkov once wrote:

> 
Back in 2014-2017, I was so inspired by Backdrop! I built websites, ported modules from Drupal 7, and contributed pull requests to core. But over time, I noticed my pull requests often sat there unreviewed. Meanwhile, technologies were moving forward, and the Backdrop community seemed stuck debating things like whether to call a button “Edit” or “Manage”—and called that UX progress.

Doesn’t this kind of situation feel familiar? I wonder if trying to get consensus on _every single issue_ is actually slowing us down. What if the community had a way to handle situations where discussions reach a stalemate—maybe by empowering someone to step in, make a decision, and move things forward? It could help avoid these moments where everyone shares their perspective, but nothing gets resolved.

I don’t mean to criticize—this is such a great community! I just think we could use a little help keeping discussions productive and focused. Let’s keep making Backdrop awesome!

**Comment by indigoxela** [comment]
@olafgrabienski  With showcasing an alternative approach I didn't mean to derail this issue. Sorry about that, it's my fault. The approach could easily simply get incorporated in the PR by @stpaultim, which currently re-introduces a regression we already [fixed in the past](https://github.com/backdrop/backdrop-issues/issues/5599). So the PR _needs work_, anyway.

There seems to be some forward and back re grey background or not, but...

We're _past feature freeze for 1.30_ and a dramatic change like a different style for the home page is unlikely to go in - and such a change requires a minor release.

However, just fixing the gap - without other display changes like moving the messages or using a substantially different style - is still possible. And it's possible in a backwards compatible way, now we have the tools available for that (since #4782 got merged). So I think, we really should fix the gap problem now, which gets reported again and again. No wonder, as this lets the home page look weird for everyone, who just installed B.

@alanmels I understand, how it's frustrating to watch such a (seemingly endless) discussion, but on the other hand proper evaluation is a substantial part of finding consensus in an Open Source project. And in the end we usually succeed. It just sometimes takes a little longer than desired.

@izmeez I agree, that the issue description didn't really keep up with the solutions discussed. I wouldn't call it "awful" - it's just out of sync.

My suggestion here: fix _one_ problem - the immediate one. The current PR introduces additional problems by trying to do too much (margin, padding...). If we fix one problem, it's easier to test and find consensus. A new style for home is only possible with the next minor in May.

**Comment by olafgrabienski** [comment]
@indigoxela I'm fine with focusing on a fix in the narrower sense, and I still hope we get it in Backdrop 1.30. Doing so, let's however continue to check what happens when messages are moved to other places. Not with the goal to improve the display, but with the goal to catch possible regressions.

**Comment by indigoxela** [comment]
> Doing so, let's however continue to check what happens when messages are moved to other places.

I did now. As already suspected - it won't be possible to fix each and every ordering, but *some* adaptions are possible.
Again by using `has()` and giving negative margin to the parent (.l-top), but only if needed.

Happy testing with https://github.com/backdrop/backdrop/pull/4978

I'm pleading, though, that you shouldn't expect perfection - the adaption for changed order and blocks removed/added to l-top already took quite a while...

**Comment by olafgrabienski** [comment]
Many thanks for updating [PR 4978](https://github.com/backdrop/backdrop/pull/4978), @indigoxela. I've had a look at the sandbox: The message on the homepage in the default position looks okay now (and much better than before). [*]

I've also moved the messages to other places and compared the sandbox to a demo site: Didn't find any visual regression. Same result when I removed the cards block, and still the same when I checked the Default layout.

So, works for me!

---

[*] Vertical alignment could be improved in a follow-up:

- When the messages are in the default position (below the Hero block), they could need a margin-top of 2rem.
- When they get moved _above_ the Hero block, margin-top should remain zero, but margin-bottom of 1.375rem would be nice.

**Comment by indigoxela** [comment]
@olafgrabienski many thanks for testing :pray: 

I just tested again myself... and found a problem when a hero block or the cards view block are used in l-top in layout templates other than boxton, but that seems feasible with further limiting scope to l-top immediately following l-wrapper.
Additional commit pushed just now.

It's really better to address the vertical alignment problem in a follow-up. Let's just try to not make things worse now. :wink: It's much trickier than expected. Like some sort of Jenga or Mikado game... :grimacing:

**Comment by olafgrabienski** [comment]
Sure, I've quickly tested again the default position and when the messages on Home are moved above the Hero block. Still works for me.

**Comment by stpaultim** [comment]
> It's much trickier than expected. Like some sort of Jenga or Mikado game... 

:-) Yes, it is.

**Comment by laryn** [comment]
If I had my druthers I'd prefer removing the grey background instead of expanding it. But if that conversation is going to slow things down I'm willing to set that aside so we can at least make this look better out of the box -- which @indigoxela's PR certainly does. I've tested it with a basic installation, which I think is the critical point. I like the use of `:has` to limit the scope of the negative padding.

**Comment by indigoxela** [comment]
> With no cards the border for messages above and below are equal. ... With cards the border for messages is not, and the background is grey like cards.

@izmeez I know what you mean. :wink: It's just - that's the "Jenga" part: whatever one changes with messages margin/padding, it breaks something else. 

That's the reason for my pleading to not expect perfection for each and every position and my suggestion to discuss this (vertical alignment) in a follow-up.
Because - as you also mentioned - who would want to move the messages between header and hero, where they'll always look odd (regardless of messages padding).

**Comment by indigoxela** [comment]
As people seem to have a hard time with the obvious margin differences (depending on messages position), I further narrowed the selector down.

Unfortunately that means, that we're back to "needs testing"... :wink:

**Comment by indigoxela** [comment]
FTR: here's a screenshot to compare various positions with the change:

![home-styles](https://github.com/user-attachments/assets/025d4fb1-ef80-4f12-ac3d-512b0f4b97bd)

In the default position the weird gap is gone.
Messages above here doesn't look that weird anymore (still not recommended).
Removing the cards block from l-top also removes the common grey background.

**Comment by jenlampton** [comment]
This is much better, but I'd still like to see matching margins around the messages for the default position. 

If we're going to use `:has` (great idea btw!) we can also use that to fix the spacing between cards and messages:
```
body.update-1-30 .l-wrapper > .l-top:has(.block-system-messages) .block-views-promoted-cards-block {
  padding-top: 0;
}
```
<img width="1636" alt="Screenshot 2025-01-09 at 12 40 12 PM" src="https://github.com/user-attachments/assets/719b1ec0-99c8-4572-8d6e-29b72d693a22" />

**Comment by indigoxela** [comment]
Oops, another PR. :wink: I moved mine out of the way - and we're back to "needs testing".

Hey folks, please test the new PR - we're slowly running out of time for 1.30, so testing should happen soon. :wave:

**Comment by indigoxela** [comment]
Lot's of people participated here. I'm trying to motivate some to do testing/review, as the 1.30 release date approaches - it would be such a shame, if this fix wouldn't go in. :pleading_face: 

@stpaultim as you're not providing the current PR, you could easily switch over to testing/review.

@izmeez the latest PR fixes the padding/margin problem that irritated you - wanna give it a try?

@olafgrabienski you've been constantly testing different and adapted approaches, I belief, the current PR will make you happy.

**Comment by olafgrabienski** [comment]
Thanks for the reminder, @indigoxela! I'll test the PR this weekend.

**Comment by alanmels** [comment]
Looks good to me! Please go ahead and close this issue, as everyone will likely style their website differently anyway. I totally understand the need for community consensus on important code-related issues that impact core functionality. However, when it comes to purely aesthetic or graphical matters, I think it would be great if we could find more efficient ways to delegate these decisions to designated individuals. 

![Screenshot 2025-01-11 at 9 56 50 AM](https://github.com/user-attachments/assets/41c69d9a-6e30-4791-b3f6-3e557f62e4ed)
![Screenshot 2025-01-11 at 9 57 33 AM](https://github.com/user-attachments/assets/6ab1a497-e2de-4fb6-bf6e-62279501ac2f)
![Screenshot 2025-01-11 at 9 58 33 AM](https://github.com/user-attachments/assets/94e845d0-82e9-48d7-ba8c-f6dc028e5fb9)

**Comment by olafgrabienski** [comment]
I've tested the new PR (default position, messages moved above the Hero block, Cards disabled, Hero disabled). Works for me!

In the default position, the spacing looks better than before when the message appears. The cards are however a bit close at the hero, when the message is dismissed (see code comment by @indigoxela). If we can't fix one situation without worsening the other before the 1.30 release, I'd suggest to leave the PR as is and to file a follow-up.

**Comment by olafgrabienski** [comment]
@izmeez – Thanks for having a look! It's a pity that the sandbox of https://github.com/backdrop/backdrop/pull/4985 isn't working at the moment.

> It may be too late for this release.

I don't think so. Testing is (always welcome but) not anymore necessary. The PR needs however a code review.

**Comment by stpaultim** [comment]
I think that all the PR sandboxes are broken at the moment. I tested locally. I think that the current PR is fine, with the exception that @indigoxela, @olafgrabienski and others have mentioned already. 

I'm relatively confident that this can be fixed in time to merge this PR. NOTE: I tried removing the 3 cards block and experience a similar issue when that block is not there. However, I do not experience this problem when moving the "messages" block to other positions. 

Screenshots:

Before doing anything.
![image](https://github.com/user-attachments/assets/7568fb44-9018-4ef3-a46f-197886a166d4)

---
Immediately after clearing the cache.
![image](https://github.com/user-attachments/assets/7e8ba188-f10e-4269-8985-683192f22159)

---
After dismissing the messages block.
![image](https://github.com/user-attachments/assets/b722449f-323c-40e7-8484-6242230d1ecf)

---
After refreshign the page.
![image](https://github.com/user-attachments/assets/31d51b67-d876-430d-876f-6fac5fc84d22)

**Comment by laryn** [comment]
I'm going to take a look at this and see if anything comes to mind to solve the final issue that has been noted. It took me a minute to figure out why the PR didn't seem to do anything when I pulled it into my local -- I guess the supplemental selectors work! ;)

**Comment by indigoxela** [comment]
> I guess the supplemental selectors work! ;)

:laughing: yes, they do. You can opt in, though, by setting `"css_update_version": "1.30",` in your basis.settings.json, or do a config_set() via devel tools. :wink:

**Comment by indigoxela** [comment]
> I'm going to take a look at this and see if anything comes to mind to solve the final issue that has been noted.

My suggestion's still to just drop that single tweak, if there's no nifty way to fix it. Even without that one tweak for padding, the PR improves home page display of messages. It would be such a shame to lose that all, because of one failed tweak. :wink:

**Comment by laryn** [comment]
Yes, I think that's the backup plan. But I'm thinking through whether there's a nifty fix first.

**Comment by laryn** [comment]
@indigoxela How does this look visually? It's a nifty fix if the gap size feels appropriate -- in line with other vertical gaps on the page.
![weird-gap-basis](https://github.com/user-attachments/assets/5384abb1-7de3-4145-ad8e-c66984bd5684)

**Comment by laryn** [comment]
I know we're really tight on time so I've just pushed this to the PR directly for testing.

**Comment by indigoxela** [comment]
Re your latest change...

You underestimate the "Jenga Effect". This fixes it for *one* constellation, but breaks it for another. If messages get moved above the hero this happens after dismissing the message:

![another-side-effect](https://github.com/user-attachments/assets/9797c959-5f8f-49de-880e-916391cf02e1)

> I know we're really tight on time so I've just pushed 

IMO that's not ideal, as this will mess the PR with last-minute-experiments, which can make it hard to decide, whether this can go in as-is or not. Worst case we introduce a regression, that requires a follow-up...

**Comment by laryn** [comment]
@indigoxela That doesn't mean I gave up. :laughing: Refiled with Jenga fix:
https://github.com/backdrop/backdrop/pull/4994

**Comment by indigoxela** [comment]
Testing https://github.com/backdrop/backdrop/pull/4994 - I belief, this is as close as we can get to perfection. :laughing: 

1. hero
2. messages
3. cards

Works, also after dismiss


1. messages
2. hero
3. cards

Works with a minimal difference in top/bottom margin of messages. Gone after dismiss.


1. messages
2. hero

(no cards) doesn't look great, but OK. That's a weird order anyway.


1. hero
2. messages

Perfect.

1. cards
2. messages

(no hero) doesn't look great, but OK. Again a weird order.

1. messages
2. cards

Perfect.

**Comment by olafgrabienski** [comment]
Same here, tested in the sandbox, works for me. @laryn – _many_ thanks for this last minute action!

**Comment by alanmels** [comment]
deep sigh...

**Comment by herbdool** [comment]
Merged into 1.x
Thanks @laryn, @indigoxela, @jenlampton for your fixes and reviews of each others' work. Thanks @wesruv for suggesting fixes. Thanks to @olafgrabienski, @izmeez, @alanmels, @stpaultim, @avpaderno, @bugfolder. Hopefully that is everyone.

**Comment by jenlampton** [comment]
Thank you all, especially @laryn and @herbdool for saving the day!
